tutorial/lesson_18_parallel_associative_reductions.cpp

```diff
@@ -97,28 +97,32 @@ int main(int argc, char **argv) {
         histogram(input(r.x, r.y)) += 1;
 
         // The factoring of associative reduction is moved into the schedule,
-        // via rfactor. To generate the same code as the manually-factored
-        // version, we do the following:
+        // via rfactor. rfactor takes as input a list of <RVar, Var> pairs,
+        // which contains list of reduction variables (RVars) to be made
+        // "parallelizable". In the intermediate, all references to this
+        // reduction variables are replaced with references to "pure" variables
+        // (the Vars). Since by construct, Vars are race-condition free, the
+        // reduction is now parallelizable across those dimensions. All reduction
+        // variables not in the list are removed from the original function and
+        // "lifted" to the intermediate.
+
+        // To generate the same code as the manually-factored version, we do the
+        // following:
         Func intm = histogram.update().rfactor({{r.y, y}});
-        // rfactor takes as input a list of <RVar, Var> pairs, which contains all
-        // reduction variables to be removed from the original function and lifted
-        // to the intermediate function. The remaining reduction variables are
-        // made 'pure' in the intermediate function, which make them race-condition
-        // free and hence are parallelizable.
-
-        // Since there is only one pair passed to rfactor, we could also write it
-        // this way:
+        // Since there is only one pair passed to rfactor, we could also write
+        // it this way:
         // Func intm = histogram.update().rfactor(r.y, y);
 
-        // Similar to the manual version, the intermediate is parallelizable
-        // across the y dimension:
+        // Here, we pass {r.y, y} as argument to rfactor to make the histogram
```

Comment:
- Here, we pass {… → We pass {…
- as *the* argument

Also, I think this sentence should go *before* the "Since there is only one pair…" aside.